src/plugins/data_views/server/routes/create_index_pattern.ts

```diff
@@ -46,51 +52,63 @@ const indexPatternSpecSchema = schema.object({
   runtimeFieldMap: schema.maybe(schema.recordOf(schema.string(), runtimeFieldSpecSchema)),
 });
 
-export const registerCreateIndexPatternRoute = (
-  router: IRouter,
-  getStartServices: StartServicesAccessor<
-    DataViewsServerPluginStartDependencies,
-    DataViewsServerPluginStart
-  >
-) => {
-  router.post(
-    {
-      path: '/api/index_patterns/index_pattern',
-      validate: {
-        body: schema.object({
-          override: schema.maybe(schema.boolean({ defaultValue: false })),
-          refresh_fields: schema.maybe(schema.boolean({ defaultValue: false })),
-          index_pattern: indexPatternSpecSchema,
-        }),
+const registerCreateDataViewRouteFactory =
+  (path: string, serviceKey: string) =>
+  (
+    router: IRouter,
+    getStartServices: StartServicesAccessor<
+      DataViewsServerPluginStartDependencies,
+      DataViewsServerPluginStart
+    >
+  ) => {
+    router.post(
+      {
+        path,
+        validate: {
+          body: schema.object({
+            override: schema.maybe(schema.boolean({ defaultValue: false })),
+            refresh_fields: schema.maybe(schema.boolean({ defaultValue: false })),
+            [serviceKey]: indexPatternSpecSchema,
+          }),
+        },
       },
-    },
-    router.handleLegacyErrors(
-      handleErrors(async (ctx, req, res) => {
-        const savedObjectsClient = ctx.core.savedObjects.client;
-        const elasticsearchClient = ctx.core.elasticsearch.client.asCurrentUser;
-        const [, , { indexPatternsServiceFactory }] = await getStartServices();
-        const indexPatternsService = await indexPatternsServiceFactory(
-          savedObjectsClient,
-          elasticsearchClient,
-          req
-        );
-        const body = req.body;
+      router.handleLegacyErrors(
+        handleErrors(async (ctx, req, res) => {
+          const savedObjectsClient = ctx.core.savedObjects.client;
+          const elasticsearchClient = ctx.core.elasticsearch.client.asCurrentUser;
+          const [, , { dataViewsServiceFactory }] = await getStartServices();
+          const indexPatternsService = await dataViewsServiceFactory(
+            savedObjectsClient,
+            elasticsearchClient,
+            req
+          );
+          const body = req.body;
 
-        const indexPattern = await indexPatternsService.createAndSave(
-          body.index_pattern as IndexPatternSpec,
-          body.override,
-          !body.refresh_fields
-        );
+          const indexPattern = await indexPatternsService.createAndSave(
+            body[serviceKey] as DataViewSpec,
+            body.override as boolean,
```

Comment:
Looks like type inference from schema got broken because of the dynamic schema approach we took here? :(